src/edit/main.js

```diff
@@ -156,11 +156,22 @@ class ProseMirror {
       domDrop: new DOMSubscription
     }
 
-    if (opts.place && opts.place.appendChild)
+    if (opts.place && opts.place.appendChild) {
+      this.root = opts.place
+
+      // loop if root is not document (in light dom)
+      // only shadowRoot has property (in shadow dom)
+      while (this.root!== document && !this.root.host) {
```

Comment:
> `opts.place` doesn't change `this.wrapper`

If `opts.place` is a function provided by the consumer, it puts `this.wrapper` in the DOM. The way your code is currently structured does not deal with this scenario correctly.